src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static dying() {
+    const dyIndex = this.alive.findIndex(animal => animal.health <= 0);
+
+    Animal.alive.splice(dyIndex, 1);
+  }
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    if (this.hidden) {
+      this.hidden = false;
+    } else {
+      this.hidden = true;
+    }
```

Comment:
```suggestion
  this.hidden = true;
```

this method should not toggle hidden value and only change it to true